merkle/sparse_merkle_tree_test.go

```diff
@@ -126,58 +139,73 @@ func TestRootAtRevision(t *testing.T) {
 }
 
 func TestRootAtUnknownRevision(t *testing.T) {
-	r, tx := getSparseMerkleTreeReaderWithMockTX(100)
-	tx.On("Commit").Return(nil)
-	tx.On("GetMerkleNodes", int64(23), mock.MatchedBy(isRootNodeOnly)).Return([]storage.Node{}, nil)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	r, tx := getSparseMerkleTreeReaderWithMockTX(mockCtrl, 100)
+	tx.EXPECT().Commit().AnyTimes().Return(nil)
+	tx.EXPECT().GetMerkleNodes(int64(23), rootNodeMatcher{}).Return([]storage.Node{}, nil)
 	_, err := r.RootAtRevision(23)
 	if err != ErrNoSuchRevision {
 		t.Fatalf("Attempt to retrieve root an non-existent revision did not result in ErrNoSuchRevision: %v", err)
 	}
 }
 
 func TestRootAtRevisionHasMultipleRoots(t *testing.T) {
-	r, tx := getSparseMerkleTreeReaderWithMockTX(100)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	r, tx := getSparseMerkleTreeReaderWithMockTX(mockCtrl, 100)
 	n1, n2 := getRandomRootNode(t, 14), getRandomRootNode(t, 15)
-	tx.On("Commit").Return(nil)
-	tx.On("GetMerkleNodes", int64(23), mock.MatchedBy(isRootNodeOnly)).Return([]storage.Node{n1, n2}, nil)
+	tx.EXPECT().Commit().AnyTimes().Return(nil)
+	tx.EXPECT().GetMerkleNodes(int64(23), rootNodeMatcher{}).Return([]storage.Node{n1, n2}, nil)
 	_, err := r.RootAtRevision(23)
 	if err == nil || err == ErrNoSuchRevision {
 		t.Fatalf("Attempt to retrieve root an non-existent revision did not result in error: %v", err)
```

Comment:
It looks like I might have cut and pasted this message from the test above. It should say that it retrieved the wrong root for the requested revision.